File: cartipay-main/Pardakht.PardakhtPay/BackEnd/Pardakht.PardakhtPay.Shared.Models/Entities/ManualTransfer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Pardakht.PardakhtPay.Shared.Models.Models;

namespace Pardakht.PardakhtPay.Shared.Models.Entities
{
    public class ManualTransfer : BaseEntity, IOwnerGuid, ITenantGuid
    {
        public DateTime CreationDate { get; set; }

        public DateTime? UpdateDate { get; set; }

        [MaxLength(70)]
        public string TenantGuid { get; set; }

        [MaxLength(70)]
        public string OwnerGuid { get; set; }

        public int CardToCardAccountId { get; set; }

        [NotMapped]
        public List<int> CardToCardAccountIds { get; set; }

        

        [MaxLength(4000)]
        public string AccountGuid { get; set; }

        public int TransferType { get; set; }

        [Column(TypeName = "decimal(18, 2)")]
        public decimal Amount { get; set; }

        //public int TransferStatus { get; set; }

        //public int BankStatementId { get; set; }

        public int TransferAccountId { get; set; }

        public string ToAccountNo { get; set; }

        public string Iban { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public int Status { get; set; }

        public int Priority { get; set; }

        public DateTime? ProcessedDate { get; set; }

        public DateTime? CancelledDate { get; set; }

        public bool ImmediateTransfer { get; set; }

        public DateTime? ExpectedTransferDate { get; set; }

        public string Comment { get; set; }

        public string CreatorId { get; set; }

        public string UpdaterId { get; set; }

        public string CancellerId { get; set; }

        public bool TransferWholeAmount { get; set; }
    }

    public enum ManualTransferStatus
    {
        Pending = 1,
        Processing = 2,
        PartialSent = 3,
        Sent = 4,
        PartialCompleted = 5,
        Completed = 6,
        Cancelled = 7,
        CancelledByBlockedAccount = 8,
        CancelledByDeletedAccount = 9,
        InsufficientBalanceOrDailyLimit = 10,
        CancelledByBlockedORDeletedAccount = 11
    }
}

File: cartipay-main/Pardakht.PardakhtPay/FrontEnd/Pardakht.PardakhtPay.AdminPortal/src/app/models/manual-transfer.ts
import { AgGridSearchArgs } from "./types";
import { GenericHelper } from "../helpers/generic";

export class ManualTransfer {

    id: number;

    bankLoginId: number;

    bankAccountId: number;

    creationDateStr: string;

    tenantGuid: string;

    ownerGuid: string;

    cardToCardAccountId: number;

    cardToCardAccountIds: number[];

    accountGuid: string;

    transferType: number;

    amount: number;

    transferAccountId: number;

    toAccountNo: string;

    iban: string;

    firstName: string;

    lastName: string;

    status: number;

    priority: number;

    processedDateStr: string;

    cancelledDateStr: string;

    immediateTransfer: boolean;

    expectedTransferDate: Date;

    expectedTransferDateStr: string;

    comment: string;

    creatorId: string;

    updaterId: string;

    cancellerId: string;

    transferWholeAmount: boolean;

    details: ManualTransferDetail[];

    constructor(data?: any) {
        GenericHelper.populateData(this, data);
    }
}

export class ManualTransferDetail {

    id: number;

    amount: number;

    transferGuid: string;

    transferId: number;

    transferNotes: string;

    trackingNumber: string;

    transferStatus: number;

    transferDate: Date;

    transferRequestDate: Date;

    bankStatementId: number;
}

export class ManualTransferSearchArgs extends AgGridSearchArgs {

    fromDate: Date;

    toDate: Date;

    toAccountNo: string;

    transferType: number;

    status: number;

    accountGuids: string[];
    
    filterModel: any;
}

export const TransferTypes = [{
    value: 1,
    key: 'MANUAL-TRANSFER.TRANSFER-TYPES.NORMAL'
},
{
    value: 2,
    key: 'MANUAL-TRANSFER.TRANSFER-TYPES.PAYA'
}, {
    value: 3,
    key: 'MANUAL-TRANSFER.TRANSFER-TYPES.SATNA'
}];

export const ManualTransferStatuses = [{
    value: 1,
    key: 'MANUAL-TRANSFER.STATUS.PENDING'
}, {
    value: 2,
    key: 'MANUAL-TRANSFER.STATUS.PROCESSING'
}, {
    value: 3,
    key: 'MANUAL-TRANSFER.STATUS.PARTIAL-SENT'
}, {
    value: 4,
    key: 'MANUAL-TRANSFER.STATUS.SENT'
}, {
    value: 5,
    key: 'MANUAL-TRANSFER.STATUS.PARTIAL-COMPLETED'
}, {
    value: 6,
    key: 'MANUAL-TRANSFER.STATUS.COMPLETED'
}, {
    value: 7,
    key: 'MANUAL-TRANSFER.STATUS.CANCELLED'
}, {
    value: 8,
    key: 'MANUAL-TRANSFER.STATUS.BLOCKED-ACCOUNT'
}, {
    value: 9,
    key: 'MANUAL-TRANSFER.STATUS.DELETED-ACCOUNT'
    }, {
        value: 10,
        key: 'MANUAL-TRANSFER.STATUS.INSUFFICIENT-BALANCE-OR-DAILY-LIMIT'
    }
];

export const TransferPriorities = [
    {
        value: 1,
        key: 'MANUAL-TRANSFER.PRIORITIES.LOW'
    }, {
        value: 2,
        key: 'MANUAL-TRANSFER.PRIORITIES.MEDIUM'
    }, {
        value: 3,
        key: 'MANUAL-TRANSFER.PRIORITIES.HIGH'
    }];

export enum TransferTypeEnum {
    Normal = 1,
    Paya = 2,
    Satna = 3
}